doc/langdef.md

```diff
@@ -167,13 +318,272 @@ type `type`, which is an expression by itself which in turn also has type
 *   `type(1) == string` evaluates to `false`
 *   `type(type(1)) == type(string)` evaluates to `true`
 
+### Protocol Buffer Data Conversion
+
+Protocol buffers have a richer range of types than CEL, so Protocol buffer data
+is converted to CEL data when read from a message field, and CEL data is
+converted in the other direction when initializing a field. In general, protocol
+buffer data can be converted to CEL without error, but range errors are possible
+in the other direction.
+
+| Protocol Buffer Type             | CEL Type                               |
+| -------------------------------- | -------------------------------------- |
+| int32, int64, sint32, sint64, sfixed32, sfixed64    | `int`               |
+| uint32, uint64, fixed32, fixed64 | `uint`                                 |
+| float, double                    | `double`                               |
+| bool, string, bytes              | same                                   |
+| enum                             | `int`                                  |
+| repeated                         | `list`                                 |
+| map<K, V>                        | `map`                                  |
+| oneof                 | options expanded individually, at most one is set |
+| message                          | same                                   |
+
+Signed integers, unsigned integers, and floating point numbers are converted to
+the singular CEL type of the same sort. The CEL type is capable of expressing
+the full range of protocol buffer values. When converting from CEL to protocol
+buffers, an out-of-range CEL value results in an error.
+
+Boolean, string, and bytes types have identical ranges and are converted without
+error.
+
+Protocol buffer enum values are converted to the corresponding `int` value.
+Protocol buffer enum fields can accept any signed 32-bit number, values outside
```

Comment:
Error is not the standard proto behavior for enums out of range. I would leave this out until we have decided about the representation of enums.

Reply:
As I understand it, there is no out-of-range for enums in proto - they must accept any 32-bit signed integer, for forward-compatibility.  This just reflect that ints outside the 32-bit signed range cannot be converted to enums.